app/js/components/orderEntry/addForm/AddForm.jsx

```diff
@@ -25,40 +24,166 @@ export class AddForm extends React.Component {
       reason: '',
       note: '',
     },
+    orders: [],
+    orderNumber: 0,
+    formType: 'Standard Dosage',
+    dosingType: '',
+    activeTabIndex: 0,
   };
+
   componentDidMount() {
     this.props.getOrderEntryConfigurations();
   }
-  handleCancel = () => {
-    // Cancel a draft drug order
+
+  handleFormTabs = (tabIndex) => {
+    this.setState({
+      activeTabIndex: tabIndex,
+    });
+  }
+
+  handleFormType = (formType) => {
+    this.setState({ formType });
+  }
+
+  handleSubmitDrugForm = () => {
+    const {
+      dose,
+      dosingUnit,
+      frequency,
+      route,
+      duration,
+      durationUnit,
+      dispensingQuantity,
+      dispensingUnit,
+      reason,
+      note,
+    } = this.state.fields;
+    this.setState({
+      orders: [
+        ...this.state.orders,
+        {
+          drugName: this.props.drugName,
+          action: "NEW",
+          dose,
+          dosingUnit,
+          frequency,
+          route,
+          duration,
+          durationUnit,
+          dispensingUnit,
+          dispensingQuantity,
+          reason,
+          note,
+          careSetting: this.props.careSetting.uuid,
+          drug: this.props.drugUuid,
+          dosingType: this.state.formType === 'Standard Dosage' ?
+            'org.openmrs.SimpleDosingInstructions' :
+            'org.openmrs.FreeTextDosingInstructions',
+          type: "drugorder",
+          orderNumber: this.state.orders.length,
+        },
+      ],
+    });
+    this.props.selectDrugSuccess('');
+    this.clearDrugForms();
+    this.props.clearSearchField();
+  }
+
+  handleEditDrugOrder = (order, drugUuid, dosingType) => {
+    this.props.selectDrugSuccess(drugUuid);
+    const {
+      dose,
+      dosingUnit,
+      frequency,
+      route,
+      duration,
+      durationUnit,
+      dispensingQuantity,
+      dispensingUnit,
+      reason,
+      note,
+    } = order;
+    this.state.orders.splice(this.state.orders.indexOf(order), 1);
+    this.setState({
+      activeTabIndex: dosingType === 'org.openmrs.SimpleDosingInstructions' ? 0 : 1,
+      fields: {
+        dose,
+        dosingUnit,
+        frequency,
+        route,
+        duration,
+        durationUnit,
+        dispensingQuantity,
+        dispensingUnit,
+        reason,
+        note,
+      },
+      ...this.state.orders,
+    });
   }
-  handleSubmitDrugForm = () => (
-    this.setState({ drugFormPopulated: true, action: "NEW" })
-  )
+
+  handleDiscardOneOrder = (order) => {
+    this.state.orders.splice(this.state.orders.indexOf(order), 1);
```

Comment:
@Betty, Splice() mutates the original array and therefore, unintentionally mutates the component state directly.

Reply:
You could look at filter(), it does not manipulate the original object.

Reply:
@flacode Why shouldn't the array be mutated? If I make a copy; how will I reconcile the two; because when deleting, it means that order is entirely not needed.